### PR TITLE
Attempt to fix https://github.com/reactjs/react-router/issues/3241

### PR DESCRIPTION
**Pull request by slorber**

### CODE REVIEW

File: modules/createBrowserHistory.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ function createBrowserHistory(options={}) {
       key = history.createKey()
 
       if (isSupported)
-        window.history.replaceState({ ...historyState, key }, null, path)
+        window.history.replaceState({ ...historyState, key }, null, window.location.href)
```
Comment (taion) [inline modules/createBrowserHistory.js]: The 3rd arg here should just be omitted, e.g.

```js
window.history.replaceState({ ...historyState, key }, null)
```

Comment (slorber) [inline modules/createBrowserHistory.js]: thanks yes I think it should work too, I'll try tomorow and edit my PR

**Comment by slorber** [comment]
@taion I've tested and yes it seems to work fine for my usecase with your suggestion so I've updated the PR

**Comment by slorber** [comment]
Just also want to point out that this code:

```
    // Automatically use the value of <base href> in HTML
    // documents as basename if it's not explicitly given.
    if (basename == null && _ExecutionEnvironment.canUseDOM) {
      var base = document.getElementsByTagName('base')[0];

      if (base) basename = _PathUtils.extractPath(base.href);
    }
```

It does not work to automatically solve my usecase so I have to configure manually the basename. Anyway not sure there's anything clever to be done here. Maybe history could automatically detect that basename has a different domain than current url and issue a meaningful error or use it as default basename?

I tried `if (base) basename = base.href;` but it's not really what we want here because actually the history basename should be set to the backend server for solving the security issue, while my page has base=frontend server.

**Comment by taion** [comment]
See the recent discussion on https://github.com/mjackson/history/issues/94. The `<base href>` support in this library is just not correct and needs to be replaced entirely (e.g. with `<base basename>` or something).

**Comment by taion** [comment]
From https://github.com/mjackson/history/issues/94#issuecomment-204649227 onward.

**Comment by slorber** [comment]
@taion @mjackson @ryanflorence I've explained my issue more in-depth in https://github.com/reactjs/react-router/issues/3241#issuecomment-209970553

Now I don't want to rush you guys but I have to put something in production in 8 days :) so I need to know if you will consider merging this fix and release 2.0.2 soon, or if I should consider publishing my own fork at least temporarily

Thanks

**Comment by taion** [comment]
@slorber Would you mind adding a regression test here? The triggers for the bug are obscure enough that is a nontrivial risk of a regression later on without such a test.

**Comment by slorber** [comment]
ok i'll try to check how this can be tested :) if you have ideas plz tell me

**Comment by taion** [comment]
Give me a sec.

**Comment by taion** [comment]
I added it: https://github.com/mjackson/history/pull/274.

**Comment by slorber** [comment]
great thanks :)

**Pull request merged**